src/internal/connector/graph/middleware.go

```diff
@@ -181,6 +182,40 @@ type RetryHandler struct {
 	Delay time.Duration
 }
 
+// Intercept implements the interface and evaluates whether to retry a failed request.
```

Comment:
Should we move retry handler middleware to its own file? Also suggest adding some tests

Reply:
Could we use chaos handler middleware to simulate graph API errors?

Reply:
> Should we move retry handler middleware to its own file?
No need, imo.  It's plenty small to fit in alongside the others.

> Also suggest adding some tests
Not entirely certain there's much gain in testing here, but I added a quick suite for coverage nonetheless.

Reply:
Sounds good, we can do chaos handler based testing later. Please create an issue. I think it would help to have this test capability longer term.